drf_extra_fields/fields.py

```diff
@@ -124,7 +124,11 @@ def get_file_extension(self, filename, decoded_file):
         # Try with PIL as fallback if format not detected due
         # to bug in imghdr https://bugs.python.org/issue16512
         if extension is None:
-            image = Image.open(io.BytesIO(decoded_file))
+            try:
+                image = Image.open(io.BytesIO(decoded_file))
+            except (OSError, IOError):
+                raise ValidationError(self.INVALID_FILE_MESSAGE)
```

Comment:
This will work most of the time but as a design there might be better solutions. If someone overrides get_file_extention this will not work. We should not let the `get_file_extension` method to do the validation.  

Right place to check if the image is valid or not is `to_internal_value` method.

We can do these operations in `to_internal_value` and assign the extension to a variable to be able to use in `get_file_extension` method.

How that sounds?

Reply:
I have a few questions about this implementation:

- `Base64ImageField` uses `Base64FieldMixin`'s `to_internal_value` method. Should we change `Base64FieldMixin`'s `to_internal_value`, or override it in `Base64ImageField` without using `super`'s `to_internal_value` (which is because we want to inject some code into the middle)?

- I agree that **"Right place to check if the image is valid or not is `to_internal_value` method."**, however, why do we need to: **"assign the extension to a variable to be able to use in `get_file_extension` method"**, don't we already have the extension, why should we call `get_file_extension` again?

I got confused. 😕

Reply:
I believe proper implementation will need a change on the base64fields' design. I will merge this as is now we can deal with this later.